### PR TITLE
Use python3 features for hex<->bytes conversions

### PR DESCRIPTION
**Pull request by hukkinj1**

### CODE REVIEW

**Comment by prusnak** [comment]
For the sake of completness - these are not Python 3 features, but Python 3.5 features, but since Python 3.4 is not supported anymore I am OK with merging this.

**Comment by hukkinj1** [comment]
Haha, yes the wording might be inaccurate there. Python 3 features as in definitely not available in Python 2. These funcs were gradually added in 3.0, 3.2 and 3.5.

File: mnemonic/mnemonic.py [hunk 1]
```diff
@@ -212,7 +211,7 @@ def check(self, mnemonic: str) -> bool:
         l = len(b)  # noqa: E741
         d = b[: l // 33 * 32]
         h = b[-l // 33 :]
-        nd = binascii.unhexlify(hex(int(d, 2))[2:].rstrip("L").zfill(l // 33 * 8))
+        nd = bytes.fromhex(hex(int(d, 2))[2:].zfill(l // 33 * 8))
```
Comment (matejcik) [inline mnemonic/mnemonic.py:214]: this is fishy
i mean, it was fishy in the preexisting code as well, but seeing `bytes.fromhex(hex(...))`...

instead: `nd = int(d, 2).to_bytes(l // 33 * 4, "big")`
also the `int(hexdigest(...), 16)` below could use a corresponding modification

here's a piece of code that does this nicer, but that's for another PR (similar changes would need to apply whenever `hexdigest` or `int(x, 16)` or `hex()` is seen
```python
# bip32:
# checksum_len = initial_entropy_len // 32
# total_len = initial_entropy_len + checksum_len
# hence total_len == checksum_len * 33
total_len = len(b)
checksum_len = l // 33
entropy_len = l - checksum_len
entropy_bits = b[: -checksum_len]
checksum_bits = b[-checksum_len:]
entropy = int(entropy_bits, 2).to_bytes(entropy_len // 8, "big")
checksum = int(checksum_bits, 2).to_bytes(checksum_len // 8, "big")
return hashlib.sha256(entropy).digest() == checksum
```

Comment (hukkinj1) [reply mnemonic/mnemonic.py]: I'm slightly confused whether you want the change in this PR or another, but changed the line you mentioned.

**Pull request merged**